_sql-and-ppl/ppl/commands/join.md

```diff
@@ -0,0 +1,218 @@
+| `joinType` | Optional | The type of join to perform. Valid values are `left`, `semi`, `anti`, and performance-sensitive types (`right`, `full`, and `cross`). Default is `inner`. |
+| `left` | Optional | An alias for the left dataset (typically a subsearch), used to avoid ambiguous field names. Specify as `left = <leftAlias>`. |
+| `right` | Optional | An alias for the right dataset (typically, a subsearch), used to avoid ambiguous field names. Specify as `right = <rightAlias>`. |
+
+### Extended syntax
+
+```sql
+join [type=<joinType>] [overwrite=<bool>] [max=n] (<join-field-list> | [left = <leftAlias>] [right = <rightAlias>] (on | where) <joinCriteria>) <right-dataset>
+```
+
+The following are examples of the extended `join` command syntax:
+
+```sql
+source = table1 | join type=outer left = l right = r on l.a = r.a table2 | fields l.a, r.a, b, c
+source = table1 | join type=left left = l right = r where l.a = r.a table2 | fields l.a, r.a, b, c
+source = table1 | join type=inner max=1 left = l right = r where l.a = r.a table2 | fields l.a, r.a, b, c
+source = table1 | join a table2 | fields a, b, c
+source = table1 | join a, b table2 | fields a, b, c
+source = table1 | join type=outer a b table2 | fields a, b, c
+source = table1 | join type=inner max=1 a, b table2 | fields a, b, c
+source = table1 | join type=left overwrite=false max=0 a, b [source=table2 | rename d as b] | fields a, b, c
+```
+
+#### Extended syntax parameters
+
+The extended `join` syntax supports the following parameters.
+
+| Parameter | Required/Optional | Description |
+| --- | --- | --- |
+| `<joinCriteria>` | Required | A comparison expression specifying how to join the datasets. Must be placed after the `on` or `where` keyword in the query. |
+| `<right-dataset>` | Required | The right dataset, which can be an index or a subsearch, with or without an alias. |  
+| `type` | Optional | The join type when using extended syntax. Valid values are `left`, `outer` (same as `left`), `semi`, `anti`, and performance-sensitive types (`right`, `full`, and `cross`). Default is `inner`. |
+| `<join-field-list>` | Optional | A list of fields used to build the join criteria. These fields must exist in both datasets. If not specified, all fields common to both datasets are used as join keys. |
+| `overwrite` | Optional | Applicable only when `join-field-list` is specified. Specifies whether fields from the right dataset with duplicate names should replace corresponding fields in the main search results. Default is `true`. |
+| `max` | Optional | The maximum number of subsearch results to join with each row in the main search. Default is `0` (unlimited). |
+| `left` | Optional | An alias for the left dataset (typically a subsearch), used to avoid ambiguous field names. Specify as `left = <leftAlias>`. |
+| `right` | Optional | An alias for the right dataset (typically, a subsearch), used to avoid ambiguous field names. Specify as `right = <rightAlias>`. |
```

Comment:
```suggestion
| `right` | Optional | An alias for the right dataset (typically, a subsearch) used to avoid ambiguous field names. Specify as `right = <rightAlias>`. |
```